tests/core/test_test.py

```diff
@@ -123,6 +124,41 @@ async def test_testing_warns_if_action_unknown(
     assert "which is not part of the training data / domain" in output
 
 
+async def test_testing_with_utilizing_retrieval_intents(
+    intent_retrieval_agent: Agent,
+    intent_retrieval_bot_test_stories: Path,
+    intent_retrieval_bot_test_results: Path,
+):
+    if not os.path.exists(intent_retrieval_bot_test_results):
+        os.mkdir(intent_retrieval_bot_test_results)
+
+    result = await rasa.core.test.test(
+        stories=intent_retrieval_bot_test_stories,
+        agent=intent_retrieval_agent,
+        e2e=True,
+        out_directory=intent_retrieval_bot_test_results,
+        disable_plotting=True,
+        warnings=False,
+    )
+    failed_stories_path = os.path.join(
+        intent_retrieval_bot_test_results, "failed_test_stories.yml"
+    )
+    with open(failed_stories_path, "r") as stream:
+        failed_stories = yaml.safe_load(stream)
+    # check that the intent is shown correctly in the failed test stories file
+    intent, full_intent = "faq", "faq/is-this-legit"
+    target_intents = [full_intent, full_intent, intent]
+    for index, story in enumerate(failed_stories["stories"]):
+        assert story["steps"][0]["intent"] == target_intents[index]
+    # check that retrieval intent for actions is retrieved correctly
+    # and only when it's needed
```

Comment:
do you also want to check that the comment is the right one?

Reply:
It looks difficult to test that, YAML readers don't extract comments, but what I'm testing there is exactly what goes to the comments. So, I think it's not necessary, but if you think we need that I can figure out and add it somehow.